### PR TITLE
Fix building with gcc-10

### PR DESCRIPTION
**Pull request by jsmolic**

### CODE REVIEW

File: app-mobilephone/smstools/smstools-3.1.21-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	newins examples/smsd.conf.easy smsd.conf
+	dodoc -r doc
+
+	systemd_dounit "${FILESDIR}"/smsd.service
+	newtmpfiles "${FILESDIR}"/smsd.tmpfiles smsd.conf
+
+}
+
+pkg_postinst() {
+	touch "${ROOT}"/var/log/smsd.log || die
```
Comment (thesamesam) [inline app-mobilephone/smstools/smstools-3.1.21-r2.ebuild:73]: EROOT instead, but we need to be careful when we're chowning in pkg_* in case it's a symlink and we're enabling something naughty. It's better to do it conditionally, not at all, or advise the user to do it.

Comment (thesamesam) [reply app-mobilephone/smstools/smstools-3.1.21-r2.ebuild:73]: Useful reference re pkg_postinst carefulness (with some opinions included too): http://michael.orlitzky.com/articles/end_root_chowning_now_%28make_pkg_postinst_great_again%29.xhtml.

Comment (jsmolic) [reply app-mobilephone/smstools/smstools-3.1.21-r2.ebuild]: Thanks, I'll read it, let's see what Soap says

Comment (jsmolic) [reply app-mobilephone/smstools/smstools-3.1.21-r2.ebuild]: Well the smsd service just fails to start due to permission problems if we don't chown in pkg_postinst...

File: app-mobilephone/smstools/smstools-3.1.21-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	default
+	if use stats; then
+		sed -i -e "s:CFLAGS += -D NOSTATS:#CFLAGS += -D NOSTATS:" \
+			"${S}/src/Makefile" || die
+	fi
+	echo "CFLAGS += ${CFLAGS}" >> src/Makefile || die
+}
+
+src_compile() {
+	cd src || die
```
Comment (SoapGentoo) [inline app-mobilephone/smstools/smstools-3.1.21-r2.ebuild:41]: ditch this, try adding `-C src` to the emake call

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+ /*
+ Print the default preference parameters.
+--- a/src/prefs.h
++++ b/src/prefs.h
+@@ -84,9 +84,9 @@ typedef struct
+   char		sdate_end[12];			/* ww end date */
+   char		stime_end[12];			/* ww end time */
+   double	time_diff;			/* Execution time */
+-  }	prefstruct;
++  } prefstruct;
```
Comment (SoapGentoo) [inline sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch:133]: whitespace

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+ 
+-  prefstruct	prefs;
++extern  prefstruct	prefs;
```
Comment (SoapGentoo) [inline sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch:136]: multiple spaces and tabs here

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
++++ b/src/globals.h
+@@ -30,8 +30,8 @@
+ #include "types.h"
+ 
+ /*----------------------- miscellaneous variables ---------------------------*/
+-char		gstr[MAXCHAR];
++extern char		gstr[MAXCHAR];
+ 
+-/*------------------------------- functions ---------------------------------*/
++/*------------------------------ functions ---------------------------------*/
```
Comment (SoapGentoo) [inline sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch:89]: fix so this doesnt show up

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-AR.patch [hunk 6]
```diff
@@ -0,0 +1,11 @@
+https://bugs.gentoo.org/725284
+--- a/configure.ac
++++ b/configure.ac
+@@ -78,6 +78,7 @@ fi
+ 
+ AC_PROG_RANLIB
+ AC_PROG_INSTALL
++m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
```
Comment (SoapGentoo) [inline sci-astronomy/weightwatcher/files/weightwatcher-1.12-AR.patch:8]: just use `AM_PROG_AR`, we're not using Automake 1.6 anymore

File: x11-plugins/asclock/asclock-2.0.12-r3.ebuild [hunk 7]
```diff
@@ -60,6 +63,6 @@ src_install() {
 	doins -r themes/*
 	dodoc README README.THEMES TODO
 	cd "${D}"/${themesdir} || die
```
Comment (SoapGentoo) [inline x11-plugins/asclock/asclock-2.0.12-r3.ebuild:65]: `ED`

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: app-mobilephone/smstools/smstools-3.1.21-r2.ebuild (hunk 1 above)
Comment (SoapGentoo) [reply app-mobilephone/smstools/smstools-3.1.21-r2.ebuild]: This is a real UX-vs-security tradeoff. I'm not entirely sure which path to take

File: app-mobilephone/smstools/smstools-3.1.21-r2.ebuild (hunk 2 above)
Comment (jsmolic) [reply app-mobilephone/smstools/smstools-3.1.21-r2.ebuild]: Done

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch (hunk 5 above)
Comment (jsmolic) [reply sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch]: Done

**Review by SoapGentoo — APPROVED** [review]

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+ /*
+ Print the default preference parameters.
+--- a/src/prefs.h
++++ b/src/prefs.h
+@@ -86,7 +86,7 @@ typedef struct
+   double	time_diff;			/* Execution time */
+   }	prefstruct;
+ 
+-  prefstruct	prefs;
++extern prefstruct	prefs;
```
Comment (SoapGentoo) [inline sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch:131]: there's still a tab in here

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+ 
+ /****** read_cat ***************************************************************
+--- a/src/fits/fitswrite.c
++++ b/src/fits/fitswrite.c
+@@ -39,6 +39,8 @@
+ #include	"fitscat_defs.h"
+ #include	"fitscat.h"
+ 
++extern int bswapflag;
++
```
Comment (SoapGentoo) [inline sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch:75]: unnecessary whitespace

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch (hunk 8 above)
Comment (jsmolic) [reply sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch]: Fixed

File: sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch (hunk 9 above)
Comment (jsmolic) [reply sci-astronomy/weightwatcher/files/weightwatcher-1.12-fno-common.patch]: Done
